### DISCUSSION TITLE
OSPSuite-R ERROR in R4.1.0

### DISCUSSION THREAD

**Discussion by yangpluszhu** [Open-Systems-Pharmacology/Forum#932] [Questions & Problems] [answered]
I am trying to use the OSPsuite using R4.1.0, following the installation guidelines. However, I get the error that the package cannot be loaded. The error message is as following:

library('ospsuite')
Loading required package: rClr
Loading the dynamic library for Microsoft .NET runtime...
Loaded Common Language Runtime version 4.0.30319.42000
Error: package or namespace load failed for ‘ospsuite’:
.onLoad failed in loadNamespace() for 'ospsuite', details:
call: rClr::clrCall(dimensionTask, "AllAvailableUnitNamesFor", enc2utf8(dimension))
error:
Then how to handle? My OSPSuite-R is v10.0.39.
Thank you for your insights!

Kind regards,
Yang

**Comment by yangpluszhu** [comment]
R version 4.1.0 (2021-05-18) -- "Camp Pontanezen"
Copyright (C) 2021 The R Foundation for Statistical Computing
Platform: x86_64-w64-mingw32/x64 (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

[Previously saved workspace restored]

> library(rClr)
Loading the dynamic library for Microsoft .NET runtime...
Loaded Common Language Runtime version 4.0.30319.42000

> library(ospsuite)
Error: package or namespace load failed for ‘ospsuite’:
 .onLoad failed in loadNamespace() for 'ospsuite', details:
  call: rClr::clrCall(dimensionTask, "AllAvailableUnitNamesFor", enc2utf8(dimension))
  error: 
> sessionInfo()
R version 4.1.0 (2021-05-18)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 17763)

Matrix products: default

locale:
[1] LC_COLLATE=Chinese (Simplified)_China.936 
[2] LC_CTYPE=Chinese (Simplified)_China.936   
[3] LC_MONETARY=Chinese (Simplified)_China.936
[4] LC_NUMERIC=C                              
[5] LC_TIME=Chinese (Simplified)_China.936    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] rClr_0.9.1

loaded via a namespace (and not attached):
[1] compiler_4.1.0 R6_2.5.0       magrittr_2.0.1 tools_4.1.0   
[5] stringi_1.6.2  stringr_1.4.0 
>

**Comment by Yuri05** [comment]
Restart R and type
`Sys.setlocale("LC_ALL","English")`

If you enter now `sessionInfo()` it should produce something like this:
```
locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                           LC_TIME=English_United States.1252    
```

Can you load the ospsuite package now?

**Comment by yangpluszhu** [comment]
Dear Solodenko,The problem still exists. The full message is as following: > Sys.setlocale("LC_ALL","English")[1] "LC_COLLATE=English_United States.1252;LC_CTYPE=English_United States.1252;LC_MONETARY=English_United States.1252;LC_NUMERIC=C;LC_TIME=English_United States.1252"> library(ospsuite)Loading required package: rClrLoading the dynamic library for Microsoft .NET runtime...Loaded Common Language Runtime version 4.0.30319.42000
Error: package or namespace load failed for ‘ospsuite’: .onLoad failed in loadNamespace() for 'ospsuite', details:  call: rClr::clrCall(dimensionTask, "AllAvailableUnitNamesFor", enc2utf8(dimension))  error: > sessionInfo() R version 4.1.0 (2021-05-18)Platform: x86_64-w64-mingw32/x64 (64-bit)Running under: Windows 10 x64 (build 17763)
Matrix products: default
locale:[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          [5] LC_TIME=English_United States.1252    system code page: 936
attached base packages:[1] stats     graphics  grDevices utils     datasets  methods   base     
other attached packages:[1] rClr_0.9.1
loaded via a namespace (and not attached):[1] compiler_4.1.0 R6_2.5.0       magrittr_2.0.1 tools_4.1.0    stringi_1.6.2  stringr_1.4.0 
I don't know how to do. The package cannot be loaded.  

--------------------------------

 

 


此致
敬礼
杨铭
上海中医药大学附属龙华医院
Regards
YANG Ming,
Longhua Hospital Affiliated to Shanghai Univesity of TCM

----- 原始邮件 -----
发件人：Juri Solodenko ***@***.***>
收件人：Open-Systems-Pharmacology/Forum ***@***.***>
抄送人：yangpluszhu ***@***.***>,  Author ***@***.***>
主题：Re: [Open-Systems-Pharmacology/Forum] OSPSuite-R ERROR in R4.1.0 (#932)
日期：2021年07月08日 19点23分


Restart R and type

Sys.setlocale("LC_ALL","English")
If you enter now sessionInfo() it should produce something like this:
locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                           LC_TIME=English_United States.1252    

Can you load the ospsuite package now?

**Reply by Yuri05** [reply]
Add new environment variable **LC_ALL** and set it to **English** (check [this tutorial](https://www.onmsft.com/how-to/how-to-set-an-environment-variable-in-windows-10) if you don't know how to do this)
![grafik](https://user-images.githubusercontent.com/25061876/124916066-8ae0cc80-dff2-11eb-96b9-3e841073b727.png)

Then **restart R** and retry. Can you load the package now?

**Comment by yangpluszhu** [comment]
Dear Solodenko,I have followed your advice. However, the package still cannot be loaded.

--------------------------------

 

 


此致
敬礼
杨铭
上海中医药大学附属龙华医院
Regards
YANG Ming,
Longhua Hospital Affiliated to Shanghai Univesity of TCM

----- 原始邮件 -----
发件人：Juri Solodenko ***@***.***>
收件人：Open-Systems-Pharmacology/Forum ***@***.***>
抄送人：yangpluszhu ***@***.***>,  Author ***@***.***>
主题：Re: [Open-Systems-Pharmacology/Forum] OSPSuite-R ERROR in R4.1.0 (#932)
日期：2021年07月08日 19点48分


Add new environment variable LC_ALL and set it to English (check this tutorial if you don't know how to do this)


Then restart R and retry. Can you load the package now?

**Reply by Yuri05** [reply]
Unzip attached file into a temporary folder, restart R and switch to that temporary folder, then execute
`source("testInitPackage.R")`
What is the output?
[testInitPackage.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/6784938/testInitPackage.zip)

**Comment by yangpluszhu** [comment]
Dear Solodenko,When I used R4.1.0-32bit, it is OK. Perhaps, the problem caused by the difference between R-32bit and R-64bit.  > library(ospsuite)Loading required package: rClrLoading the dynamic library for Microsoft .NET runtime...Loaded Common Language Runtime version 4.0.30319.42000
> sessionInfo() R version 4.1.0 (2021-05-18)Platform: i386-w64-mingw32/i386 (32-bit)Running under: Windows 10 x64 (build 17763)
Matrix products: default
locale:[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252[4] LC_NUMERIC=C                           LC_TIME=English_United States.1252    system code page: 936
attached base packages:[1] stats     graphics  grDevices utils     datasets  methods   base     
other attached packages:[1] ospsuite_10.0.40 rClr_0.9.1      
loaded via a namespace (and not attached):[1] compiler_4.1.0 magrittr_2.0.1 tools_4.1.0    stringi_1.6.2  stringr_1.4.0  

--------------------------------

 

 


此致
敬礼
杨铭
上海中医药大学附属龙华医院
Regards
YANG Ming,
Longhua Hospital Affiliated to Shanghai Univesity of TCM

----- 原始邮件 -----
发件人：Juri Solodenko ***@***.***>
收件人：Open-Systems-Pharmacology/Forum ***@***.***>
抄送人：yangpluszhu ***@***.***>,  Author ***@***.***>
主题：Re: [Open-Systems-Pharmacology/Forum] OSPSuite-R ERROR in R4.1.0 (#932)
日期：2021年07月08日 19点48分


Add new environment variable LC_ALL and set it to English (check this tutorial if you don't know how to do this)


Then restart R and retry. Can you load the package now?

**Comment by ReneGeci** [comment]
@yangpluszhu did u find a solution for this problem? I am experiencing the same issue unfortunately...

**Comment by Yuri05** [comment, marked as answer]
Seems to be a bug in recent R-Studio versions. s. https://github.com/Open-Systems-Pharmacology/OSPSuite-R/issues/729#issuecomment-1025675333
